net/Socket.cpp

```diff
@@ -77,6 +77,7 @@ bool StreamSocket::simulateSocketError(bool read)
     return false;
 }
 
+#if !MOBILEAPP && ENABLE_SSL
```

Comment:
ENABLE_SSL shouldn't be needed here (because SslStreamSocket is only defined with ENABLE_SSL).

But I'm surprised we need this condition here at all, because the declaration of simulateSocketError and its usage is not guarded with `#if !MOBILEAPP`. So I expect we can remove this condition and it wouldn't change anything.